modules/cli/src/main/scala/scala/cli/launcher/LauncherCli.scala

```diff
@@ -65,4 +69,34 @@ object LauncherCli {
     sys.exit(exitCode)
   }
 
+  def resolveNightlyScalaCliVersion(
+    cache: FileCache[Task],
+    scalaParameters: ScalaParameters
+  ): String = {
+
+    val snapshotRepoUrl =
+      s"https://oss.sonatype.org/content/repositories/snapshots/org/virtuslab/scala-cli/cli_${scalaParameters.scalaBinaryVersion}/"
+    val artifact = Artifact(snapshotRepoUrl).withChanging(true)
+    val res = cache.logger.use {
+      try cache.withTtl(0.seconds).file(artifact).run.unsafeRun()(cache.ec)
+      catch {
+        case NonFatal(e) => throw new Exception(e)
+      }
+    }
+
+    res match {
+      case Left(_) =>
+        System.err.println("Unable to find nightly Scala CLI version")
+        sys.exit(1)
+      case Right(f) =>
+        val snapshotRepoPage = os.read(os.Path(f, Os.pwd))
+        val rawVersions      = coursier.CoursierUtil.rawVersions(snapshotRepoUrl, snapshotRepoPage)
+        val versions         = rawVersions.map(Version(_))
+        val nightlyVersion   = versions.max
+
+        nightlyVersion.repr
```

Comment:
Maybe let's check that `versions` is non-empty, just in case, and print a better error message if it is:
```suggestion
        if (versions.isEmpty)
          sys.error(s"No versions found in $snapshotRepoUrl (locally at $f)")
        else
          versions.max.repr
```